src/web/signer.ts

```diff
@@ -77,25 +73,13 @@ export class TurboWebArweaveSigner implements TurboDataItemSigner {
     };
   }
 
-  // NOTE: this might be better in a parent class or elsewhere - easy enough to leave in here now and does require specific environment version of crypto
-  async generateSignedRequestHeaders() {
-    const nonce = randomBytes(16).toString('hex');
-    const buffer = Buffer.from(nonce);
-    const signature = await this.signer.sign(buffer);
-
-    // for arconnect, we need to make sure we have the public key
-    if (
-      this.signer.publicKey === undefined &&
-      this.signer instanceof ArconnectSigner
-    ) {
-      await this.signer.setPublicKey();
-    }
-    const publicKey = toB64Url(this.signer.publicKey);
+  public async generateSignedRequestHeaders(): Promise<TurboSignedRequestHeaders> {
+    await this.setPublicKey();
+    return super.generateSignedRequestHeaders();
+  }
 
-    return {
-      'x-public-key': publicKey,
-      'x-nonce': nonce,
-      'x-signature': toB64Url(Buffer.from(signature)),
-    };
+  public async signTx(dataToSign: Uint8Array): Promise<Uint8Array> {
```

Comment:
Nit: this function is called `signTx` but it seems like it's signing any arbitrary data blob. Might want to either be more pedantic with the input argument (and convert it to bytes yourself) OR rename the function. This also doesn't seem scalable for large data items for which we'd need a stream.

Reply:
939e52c